HeliosCLI/Makefile

```diff
@@ -34,6 +34,13 @@ ifneq ($(INCLUDES),)
     CFLAGS+=$(INCLUDES)
 endif
 
+# Define ROOT_OUTPUT_DIR and FINAL_TARGET
+ROOT_OUTPUT_DIR ?= ../output
+FINAL_TARGET = $(ROOT_OUTPUT_DIR)/helios_cli
+
+# Build directory for object files - default to ./build if not specified
+BUILD_DIR ?= ./build

```

Comment:
I don't see how this separates the build files for each platform, you need dynamic build dir which is generated from the platform being built.  As far as I can see you have the exact same problem as before. Did you actually try building all? Maybe I'm just missing something here